src/main/src/main/java/org/geoserver/config/util/XStreamPersister.java

```diff
@@ -2291,4 +2278,39 @@ public Object doUnmarshal(Object result,
             return obj;
         }
     }
+    
+    /**
+     * Converter for ServiceInfo class
+     */
+    public class ServiceInfoConverter extends AbstractReflectionConverter {
+    
+        public ServiceInfoConverter() {
+            super(ServiceInfo.class);
+        }
+        public ServiceInfoConverter(Class<? extends ServiceInfo> clazz) {
+            super(clazz);
+        }
+        
+        public Object doUnmarshal(Object result,
+                HierarchicalStreamReader reader, UnmarshallingContext context) {
+            ServiceInfoImpl obj = (ServiceInfoImpl) super.doUnmarshal(result, reader, context);
+            
+            return obj;
+        }
+        @Override
+        protected Object instantiateNewInstance(HierarchicalStreamReader reader,
+                UnmarshallingContext context) {
+            Object emptyObject = super.instantiateNewInstance(reader, context);
```

Comment:
This is where the bug is actually fixed, can we get a javadoc or something to show how important this logic is